Yoast/wordpress-seo
# Replace usage of WPSEO_Extension_Manager in favor of WPSEO_Addon_Manager

## Summary

This PR can be summarized in the following changelog entry:

* [not-userfacing] Replaces usage of WPSEO_Extension_Manager in favor of WPSEO_Addon_Manager.

## Relevant technical choices:

* Added a helper function called `has_valid_subscription` to the `WPSEO_Addon_Manager` that gets the subscription via a slug and checks if the subscription is expired.
* Create contants for the addon slugs in `WPSEO_Addon_Manager`. This is so we can access them seperately from the protected $addons array, not having to know the key.
* `WPSEO_Extension_Manager` was used in the following locations:
	* `admin/class-admin.php` (`WPSEO_Admin`) -> `add_action_link`: the premium activated check.
	* `admin/class-extensions.php` (`WPSEO_Extensions`) -> `is_valid`: extensions activated check.
	* `admin/class-yoast-form.php` (`Yoast_Form`) -> `admin_sidebar`: the premium activated check.
	* `admin/views/licenses.php` -> Now uses an array with the slug constants as key.

## Test instructions

This PR can be tested by following these steps:

These changes can only really be checked in premium. But the code lives in free.
* Create a test branch and merge this branch in there before you test (see https://github.com/Yoast/Wiki/wiki/Merging-Free-into-Premium for some more info on how).
* Be sure to install the add-ons!
* Also, after checking the things below, for some it would be nice to check if it detects the difference if a license is not active anymore. Please ask Herre, he can change this for `local.wordpress.test`.

The following places where affected:

### WPSEO_Admin
* Go to the WordPress plugins page.
* It should not have the `FAQ` and `Premium Support` links when you have an activated Premium.
<img width="583" alt="class_admin change" src="https://user-images.githubusercontent.com/35524806/54687709-812eb480-4b1c-11e9-871b-b1604df866c6.png">


### WPSEO_Extensions
* Go to the Yoast SEO Dashboard.
* It should not show the problem notices `You are not receiving updates or support!`.
<details><summary>Notices screenshot</summary>
<img width="903" alt="Screenshot 2019-03-18 at 12 17 28" src="https://user-images.githubusercontent.com/19681708/54526699-d382a080-4977-11e9-81c3-4c29efcd77e2.png">
</details>

### Yoast_Form
* Go to the Yoast SEO Dashboard.
* It should not have a sidebar.
<details><summary>Sidebar screenshot</summary>
<img width="370" alt="Screenshot 2019-03-18 at 12 20 55" src="https://user-images.githubusercontent.com/19681708/54526833-4c81f800-4978-11e9-81fe-c7584a8d931a.png">
</details>

### Yoast SEO Dashboard -> Premium page
* Go to the Yoast SEO Dashboard -> Premium page.
* It should indicate `ACTIVATED` instead of `NOT ACTIVATED` for Premium and the addons.
<details><summary>Licenses screenshot</summary>
<img width="1656" alt="licenses view change" src="https://user-images.githubusercontent.com/35524806/54689337-99540300-4b1f-11e9-90d7-24508670e0b2.png">
</details>

## UI changes

## Quality assurance


Fixes #12446
